docs/rules/prefer-t-throws.md

```diff
@@ -0,0 +1,96 @@
+# Prefer using `t.throws()` or `t.throwsAsync()` over try/catch
+
+This rule will enforce the use of `t.throws()` or `t.throwsAsync()` when possible.
+
+## Fail
+
+```js
+const test = require('ava');
+
+test('some test', async t => {
+	try {
+		await throwingFunction();
+		t.fail();
+	} catch (error) {
+		t.is(error.message, 'Unicorn overload');
+	}
+});
+```
+
+```js
+const test = require('ava');
+
+test('some test', async t => {
+	try {
+		await potentiallyThrowingFunction();
+		await anotherPromise;
+		await timeout(100, 'Unicorn timeout');
+		t.fail();
+	} catch (error) {
+		t.ok(error.message.startsWith('Unicorn'));
+	}
+});
+```
+
+```js
+const test = require('ava');
+
+test('some test', async t => {
+	try {
+		synchronousThrowingFunction();
+		t.fail();
+	} catch (error) {
+		t.is(error.message, 'Missing Unicorn argument');
+	}
+});
+```
+
+## Pass
+
+```js
+const test = require('ava');
+
+test('some test', async t => {
+	const error = await t.throwsAsync(asyncThrowingFunction());
+	t.is(error.message, 'Unicorn overload');
```

Comment:
This is not the recommended way to test the error message.

Reply:
That is also not the recommended way. Please read the docs for `t.throwsAsync`. It accepts a `message` option.